pkg/plugins/runtime/k8s/controllers/cni_taint_controller.go

```diff
@@ -185,7 +179,7 @@ func podEvents(cniApp string) predicate.Funcs {
 			return filterPods(deleteEvent.Object, cniApp)
 		},
 		UpdateFunc: func(updateEvent event.UpdateEvent) bool {
-			return filterPods(updateEvent.ObjectNew, cniApp)
+			return filterPods(updateEvent.ObjectNew, cniApp) && filterPods(updateEvent.ObjectOld, cniApp)
```

Comment:
I think it's quite valid to only check `ObjectNew` since we might only care about the new state, it was just an explanation as to why an Object with no NodeName might make it to the mapper.

Reply:
There's no way to add predicates _after_ the mapper so I think it all has to be checked there.

Reply:
So you'd leave the filtering in the mapper and only check the new object here? Theoretically there are more events between pod being created and ready but you rightly pointed out that we only care about the new state here.

Reply:
> There's no way to add predicates after the mapper so I think it all has to be checked there.

So I'm reverting the last commit and modifying the comment.

Reply:
The mapping/reconciling is a bit weird. `Reconcile` is called level-based so it's called once for an `UpdateEvent` and the cache will have `ObjectNew` in it. For mappers however, an `UpdateEvent` triggers two calls to the mapper, which is why all the checking logic has to be in the mapper.

Reply:
And also, it may be more correct to not use `filterPods` in the `Handler` at all but instead in the mapper. Otherwise, because `podEvents` correctly checks only `ObjectNew`, the mapper may be called with an `ObjectOld` that _doesn't_ pass `filterPods` and may trigger an extra reconciliation.

Reply:
I've put it back to the mapper.